pkg/cmd/templatefuncs.go

```diff
@@ -28,9 +28,14 @@ type ioregData struct {
 	value map[string]any
 }
 
-// needsQuoteRx matches any string that contains non-printable characters,
-// double quotes, or a backslash.
-var needsQuoteRx = regexp.MustCompile(`[^\x21\x23-\x5b\x5d-\x7e]`)
+var (
+	// lineEndingRx matches line endings.
+	lineEndingRx = regexp.MustCompile(`(?m)(?:\r\n|\r|\n)`)
```

Comment:
I *think* that `$` at the end of this would be advised, because `(?m)` makes it so that `$` matches end of line, not just end of text (for which one should always be using `\z`). It’s worth a local test at least.

Reply:
I did a quick test with this, changing this line to:

```go
	lineEndingRx = regexp.MustCompile(`(?m)$(?:\r\n|\r|\n)`)
```

(note the `$` in the middle)

and also with

```go
	lineEndingRx = regexp.MustCompile(`(?m)(?:\r\n|\r|\n)$`)
```

(note the `$` at the end)

Both approaches failed the existing tests. I think this is because the zero-width look-ahead `$` only knows about UNIX and Windows line endings, and so misbehaves when it encounters the old macOS `\r`.

Maybe we need to do something else?

Reply:
Could you not use `(?m)(?:\r\n$|\r$|\n$)`?

Reply:
I suspect that would fail as well. It’s been a *long* time since I’ve had to do line-ending checks in RE, and the Go RE engine (RE2) is subtly and intentionally different than PCRE, Oniguruma, or the Vim RE, so my guess is that you can *either* match `\n` *or* `$`, not both.